docs/config_docs.rst

```diff
@@ -0,0 +1,143 @@
+Configuration documentation
+===========================
+
+The library can generate (and check) the documentation for model fields which are being
+configured, containing information about the names of the fields, possible values,
+default values, and a short description. This helps clients determine what can be
+configured with the help of the library and how.
+
+
+Setup
+"""""
+
+Start by defining the target directory of the docs and the template that will be used
+to render them. For example:
+
+::
+
+    DJANGO_SETUP_CONFIG_TEMPLATE_NAME = "configurations/config_doc.rst"
+    DJANGO_SETUP_CONFIG_DOC_DIR = "docs/configuration"
+
+Next, create a template:
+
+::
+
+    {% block link %}{{ link }}{% endblock %}
+
+    {% block title %}{{ title }}{% endblock %}
+
+    Settings Overview
+    =================
+
+    Enable/Disable configuration:
+    """""""""""""""""""""""""""""
+
+    ::
+
+        {% spaceless %}
+        {{ enable_settings }}
+        {% endspaceless %}
+
+    Required:
+    """""""""
+
+    ::
+
+        {% spaceless %}
+        {% for setting in required_settings %}{{ setting }}
+        {% endfor %}
+        {% endspaceless %}
+
+    All settings:
+    """""""""""""
+
+    ::
+
+        {% spaceless %}
+        {% for setting in all_settings %}{{ setting }}
+        {% endfor %}
+        {% endspaceless %}
+
+    Detailed Information
+    ====================
+
+    ::
+
+        {% spaceless %}
+        {% for detail in detailed_info %}
+        {% for part in detail %}{{ part|safe }}
+        {% endfor %}{% endfor %}
+        {% endspaceless %}
+
+You're free to choose a different layout, of course, but this should give you an idea
+of what's available, and some of the nuisances (like spacing) you might run into.
+
+The value for ``link`` is automatically create based on the ``file_name`` you defined
+in the first step. Similarly, ``enable_settings`` is automatically created. Values for
+the other template variables are determined based on the model your client is configuring,
+and a related settings model you need to define.
+
+Generally, when adding support for configuring a model ``FooConfiguration`` with
+``django-setup-configuration``, you will have a class ``FooConfigurationStep(BaseConfigurationStep)``
+that carries out the configuration. In the same file, define a class with information about
+the fields from ``FooConfiguration`` that are required, included, or excluded, and
+meta-information required for creating the docs. For example:
+
+
+::
+
+    from django-setup-configuration import ConfigSettingsModel
+
+    class FooConfigurationSettings(ConfigSettingsModel):
+        model = FooConfiguration
+        display_name = "Foo Configuration"
+        namespace = "FOO"
+        required_fields = (
+            "bar",
+            "baz",
+        )
+        included_fields =  required_fields + (
+            "foobar",
+        )
+        excluded_fields = (
+            "bogus",
+        )
```

Comment:
I think these are no longer part of the code, only `required_settings` and `optional_settings`